### PR TITLE
Omit zero price products

### LINKED ISSUE #265: Request to omit zero cost products from Pinterest feed
### Is your feature request related to a problem? 
(https://wordpress.org/support/topic/request-to-omit-zero-cost-products-from-pinterest-feed/)

 > I have about 100 products in my shop that are free downloads, so the product price is $0.00. This is a lead magnet to get my customers to my online shop to buy digital products. On Pinterest side, it always shows an Error 113 for the 101 free products, as Pinterest does not accept 0 values for price.
> 
> Error 113: “Some items have price value formatting errors in product metadata, so those items will not be published”

#### How to reproduce the problem
Add products to WooCommerce with price = 0.00

### Describe the solution you'd like 

> If possible, would you make a change to exclude any products with zero cost from the feed in a future release? Many in my industry follow these same practices of offering zero cost items. So, this is more of an enhancement request. Perhaps an option to exclude products in the future using the settings?

### PR DESCRIPTION
**Pull request by alexiglesias31**
### Changes proposed in this Pull Request:

Closes #265 .

Exclude $0.00 price products on feed generation.
---
**Pinterest** does not support ingesting $0.00 price products.

### Detailed test instructions:
1. Add a product with 0.00 price
2. Verify that is not included in the feed

### Changelog entry

> Update - Omit zero price products from the feed file.

### CODE REVIEW

File: src/ProductsXmlFeed.php [hunk 1]
```diff
@@ -73,12 +73,16 @@ public static function get_xml_footer() {
 	/**
 	 * Returns the Item's XML for the given product.
 	 *
-	 * @param WC_Product $product The product to print the XML for.
+	 * @param WC_Product $product  The product to print the XML for.
 	 *
-	 * @return string
+	 * @return string|null
```
Comment (ksere) [inline src/ProductsXmlFeed.php]: Prefer keeping 1 single return type. 
And check for empty string where needed

**Review by ksere — CHANGES_REQUESTED** [review]
Small comment on a method's return type

File: src/ProductSync.php [hunk 2]
```diff
@@ -362,15 +372,23 @@ public static function handle_feed_generation() {
...
 
 			for ( self::$current_index; ( self::$current_index < $products_count ); self::$current_index++ ) {
 
 				$product_id = $product_ids[ self::$current_index ];
 
-				self::$iteration_buffer .= ProductsXmlFeed::get_xml_item( wc_get_product( $product_id ) );
-				self::$iteration_buffer_size++;
+				$xml_item = ProductsXmlFeed::get_xml_item( wc_get_product( $product_id ) );
+
+				if ( $xml_item ) {
```
Comment (ksere) [inline src/ProductSync.php]:
```suggestion
				if ( '' !== $xml_item ) {
```

**Review by ksere — APPROVED** [review]

**Comment by budzanowski** [comment]
Rebased with `develop` to fix merge conflicts.

**Review by budzanowski — APPROVED** [review]

**Pull request merged**